### ISSUE TITLE
CocoaPods installation is painful

### ISSUE THREAD

**Issue by khandaresuyash** [CocoaPods/CocoaPods#12640]
# 

## What did you do?
Factory reset my M1 Macbook AIR. Tried installing cocoapods with:
`sudo gem install cocoapods`

## What did you expect to happen?
I expected to install cocapods on my device.

## What happened instead?
1. Installing cocoa pods doesn't have any progress indicator and took long time.
2. I was greeted with
    `ERROR:  Error installing cocoapods:
	The last version of drb (>= 0) to support your Ruby & RubyGems was 2.0.6. Try installing it with `gem install drb -v 2.0.6` and then running the current command again`.
3. After running the suggested command and `install cocapods` it greeted me with new dependency issue
   `ERROR:  Error installing cocoapods:
	The last version of activesupport (>= 5.0, < 8) to support your Ruby & RubyGems was 6.1.7.9. Try installing it with `gem install activesupport -v 6.1.7.9` and then running the current command again`.
4. This happened over and over for multiple  different dependencies/packages.
<img width="1440" alt="Screenshot 2024-10-18 at 1 51 09 AM" src="https://github.com/user-attachments/assets/b48d73ae-406a-4568-9205-216957eebe2b">

**Comment by Rufy86** [comment]
same problem

**Comment by jshier** [comment]
You should never install any Ruby gem using `sudo gem install`. The system Ruby is privileged and shouldn't be touched. Instead, you should use a Ruby manager like RVM or rbenv, which can manage your Ruby versions and gems locally in an unprivileged fashion. (If you try to install RVM, install from their head or master branch, they don't seem to do stable releases anymore.)

**Comment by theJohnnyBrown** [comment]
Same problem here.

> You should never install any Ruby gem using `sudo gem install`. The system Ruby is privileged and shouldn't be touched. Instead, you should use a Ruby manager like RVM or rbenv, which can manage your Ruby versions and gems locally in an unprivileged fashion. (If you try to install RVM, install from their head or master branch, they don't seem to do stable releases anymore.)

Then why is this the instructions given on https://cocoapods.org?

**Comment by monfresh** [comment]
@theJohnnyBrown That's a great question. I don't know the reason why the official instructions recommend using `sudo` and using the system Ruby that comes preinstalled on macOS, but @jshier is right. That is not recommended at all, and I have been repeating the same thing in various issues in this repo. Unfortunately, Cocoapods is only in maintenance mode now as mentioned in this blog post: https://blog.cocoapods.org/CocoaPods-Support-Plans/

If the Cocoapods maintainers are open to it, I'd be happy to submit a pull request to update the official instructions.

In the meantime, I recommend reading these guides that I wrote that explain: 

- [Why You Shouldn't Use the System Ruby to Install Gems on a Mac](https://www.moncefbelyamani.com/why-you-shouldn-t-use-the-system-ruby-to-install-gems-on-a-mac/)
- [Why You Should Never Use sudo to Install Ruby Gems](https://www.moncefbelyamani.com/why-you-should-never-use-sudo-to-install-ruby-gems/)
- [How to properly install Ruby on a Mac](https://www.moncefbelyamani.com/how-to-install-xcode-homebrew-git-rvm-ruby-on-mac/)
- [Why you're getting the error 'drb requires Ruby version >= 2.7.0'](https://www.rubyonmac.dev/drb-requires-ruby-version-2-7-0-current-ruby-version-is-2-6-10)

**Comment by Avadhamcodr** [comment]
I found a solution that worked for me. You can try creating a new project and then replacing the necessary code and files with those from your current setup. This resolved the issue for me.

**Comment by amorde** [comment]
Yeah this has come up before (specifically the system ruby thing) and I'd love to change the docs, I just haven't gotten to it since I'm basically doing the bare minimum to keep the project working. If anyone would like to submit changes to the guides to discuss using rvm, rbenv etc. that would be great

**Comment by amorde** [comment]
Opened up a PR here https://github.com/CocoaPods/guides.cocoapods.org/pull/236

Open to any feedback

**Comment by CodingMarkus** [comment]
Don't use Apple's Ruby at all.

Install MacPorts: https://www.macports.org/install.php

Then open a Terminal:

sudo port install ruby32
sudo /opt/local/bin/gem install cocoapods

There's now a `/opt/local/bin/pod` for you.

Once MacPorts is installed, `/opt/local/bin/` is added to your PATH, so you can just call it with `pod`. You only need to use the full path for the `gem` call above to make sure that the MacPorts `gem` is being used and not the one of macOS.

@jshier  
> Instead, you should use a Ruby manager like RVM or rbenv, which can manage your Ruby versions and gems locally in an unprivileged fashion.

Installing any directly executable code unprivileged means that any malicious tool on your computer can simply modify these files and replace them with viruses, Trojans, crypto-miners and other malware. What would stop them? That's why, on traditional UNIX/Linux systems, binaries in /bin or /usr/bin or even /local/user/bin are only writable by root, even if they can be executed by anyone and usually only perform harmless operations.

MacPort follows this philosophy and ensures that all executables in /opt/local/bin are writable only by root, so unless you give malware root permission, it cannot modify your binaries.

This also makes it safe to run these tools with `sudo`, because if you run any binary with `sudo` that your user has write access to, a malware might have replaced the binary code with something else, and if you now run that tool with `sudo`, you've just given your malware root access.

**Comment by begreatfulforreal** [comment]
@CodingMarkus you saved my life :) !!

**Comment by nordbergr** [comment]
I used home-brew instead of the documentation from [https://cocoapods.org](https://cocoapods.org) and now `flutter doctor` is reporting all is good 👍
